### PR TITLE
#2026: Improve performance of building "Salable Quantity" on Product …

### PR DESCRIPTION
**Pull request by mcspronko**
Work In Progress
---

Current state of the Salable Quantity Column with the performance improvement:
Screen Shot 2019-03-09 at 09 43 01

Salable Quantity uses:
* Inventory Stock indexer tables 
* Inventory Reservation
* TBD Min Qty per product
* TBD Is In Stock legacy stock item

To be added:
* Min Qty configuration value for salable_quantity calculation.

### Fixed Issues (if relevant)
1. https://github.com/magento-engcom/msi/issues/2026: Improve performance of building "Salable Quantity" on Product Grid

### Manual testing scenarios (*)
1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/InventorySalesAdminUi/Model/ResourceModel/GetSalableQuantityData.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function execute(int $stockId, array $skus): array
+    {
+        $stockItemTableName = $this->stockIndexTableNameResolver->execute($stockId);
+
+        $subSelect = $this->resource->getConnection()->select()
+            ->from(
+                ['ir' => 'inventory_reservation'],
+                [
+                    new \Zend_Db_Expr('SUM(ir.quantity) AS quantity'),
```
Comment (maghamed) [inline app/code/Magento/InventorySalesAdminUi/Model/ResourceModel/GetSalableQuantityData.php]: Don't use ZF1 constructions

File: app/code/Magento/InventorySalesAdminUi/Model/ResourceModel/GetSalableQuantityData.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            if ($connection->isTableExists($stockItemTableName)) {
+                return $connection->fetchAll($select) ?: null;
+            }
+
+            return null;
+        } catch (\Exception $e) {
+            throw new LocalizedException(__(
+                'Could not receive Stock Item data'
+            ), $e);
+        }
```
Comment (maghamed) [inline app/code/Magento/InventorySalesAdminUi/Model/ResourceModel/GetSalableQuantityData.php:99]: looks like this try/catch is excessive as neither isTableExists nor fetchAll throws an exception

File: app/code/Magento/InventorySalesAdminUi/Model/GetSalableQuantityDataBySkus.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     * @param array $skus
+     * @return array
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \Magento\InventoryConfigurationApi\Exception\SkuIsNotAssignedToStockException
+     */
+    public function execute(array $skus): array
+    {
+        $stocks = $this->stockRepository->getList($this->searchCriteriaBuilder->create())->getItems();
```
Comment (maghamed) [inline app/code/Magento/InventorySalesAdminUi/Model/GetSalableQuantityDataBySkus.php]: ideally is to process just stocks to which at least one of those SKUs assigned to , but for a small number of stocks that should not be a big problem

**Review by maghamed — CHANGES_REQUESTED** [review]
Please make static tests pass

**Comment by ishakhsuvarov** [comment]
Hi @mcspronko 
Are you still interested in this one? There are still some review comments and TBD points left

**Comment by ishakhsuvarov** [comment]
@mcspronko I'm closing this PR for now. Please feel free to reopen if you have time to check out review comments and TBD items from the description.

**Pull request closed without merging**